photon_installer/installer.py

```diff
@@ -926,10 +926,28 @@ def _setup_install_repo(self):
     def _install_additional_rpms(self):
         rpms_path = self.install_config.get('additional_rpms_path', None)
 
-        if not rpms_path or not os.path.exists(rpms_path):
+        rpm_dir = []
+        if not rpms_path or len(rpm_dir) == 0:
             return
 
-        if self.cmd.run(['rpm', '--root', self.photon_root, '-U', rpms_path + '/*.rpm']) != 0:
+        if isinstance(rpms_path, list):
+            rpm_dir = [d for d in rpm_dir if len(d) > 0
+                       and os.path.exists(d)]
+        else:
+            if os.path.exists(rpms_path):
+                rpm_dir.append(rpm_dir)
+            else:
+                return
+        
+        cmd_err = 0
+        for rdir in rpm_dir:
+            try:
+                # best effort for each dir in list
+                cmd_err = self.cmd.run(['rpm', '--root', self.photon_root, '-U', rdir + '/*.rpm'])
+            except FileNotFoundError as _:
```

Comment:
Same here, don't ignore erroneous situations.

Small optimization here can be, passing all rpm dirs at once to `rpm --root ... -U ...`
Like:
```
rpm --root ... -U path1/*.rpm path2/*.rpm ...
```

An rpm in path1 might need an rpm from path2 as a requirement, so it will get satisfied.

Reply:
Actually we should use `tdnf` so it can resolve all dependencies.